src/Entity/Carrinho.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace App\Entity;
+
+use App\Repository\CarrinhoRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: CarrinhoRepository::class)]
+class Carrinho
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\OneToOne(inversedBy: 'carrinho', cascade: ['persist', 'remove'])]
+    private ?Cliente $cliente_id = null;
+
+    #[ORM\ManyToOne(inversedBy: 'carrinhos')]
+    private ?User $usuario_id = null;
```

Comment:
quando você fizer new no Carrinho o usuário vai ser obrigatório?


```suggestion
    private User $usuario_id;
```

Reply:
Todos os campos que são obrigatórios não podem ser nullable